flow/surface_frame.cc

```diff
@@ -24,7 +25,7 @@ SurfaceFrame::SurfaceFrame(sk_sp<SkSurface> surface,
   FML_DCHECK(submit_callback_);
   if (surface_) {
     canvas_ = surface_->getCanvas();
-  } else {
+  } else if (display_list_fallback) {
```

Comment:
Should this then have a final "else" that notifies someone that we gave up on canvas/recorder? If that can't happen, do we need the if part of this else-if?